src/components/timeline/TrackThread.js

```diff
@@ -217,21 +221,26 @@ export default explicitConnect<OwnProps, StateProps, DispatchProps>({
     const selectors = getThreadSelectors(threadIndex);
     const selectedThread = getSelectedThreadIndex(state);
     const committedRange = getCommittedRange(state);
+    const selectedCallNodeIndex =
+      threadIndex === selectedThread
+        ? selectors.getSelectedCallNodeIndex(state)
+        : null;
     return {
       filteredThread: selectors.getFilteredThread(state),
       fullThread: selectors.getRangeFilteredThread(state),
       callNodeInfo: selectors.getCallNodeInfo(state),
-      selectedCallNodeIndex:
-        threadIndex === selectedThread
-          ? selectors.getSelectedCallNodeIndex(state)
-          : -1,
+      selectedCallNodeIndex,
       unfilteredSamplesRange: selectors.unfilteredSamplesRange(state),
       interval: getProfileInterval(state),
       rangeStart: committedRange.start,
       rangeEnd: committedRange.end,
       categories: getCategories(state),
       timelineType: getTimelineType(state),
       hasFileIoMarkers: selectors.getFileIoMarkerIndexes(state).length !== 0,
+      samplesSelectedStates:
+        selectedCallNodeIndex !== null
```

Comment:
It looks like there's no test in this PR, but actually I got the idea for this check because a snapshot was failing.